src/game/boe.main.cpp

```diff
@@ -387,6 +387,21 @@ using Key = sf::Keyboard::Key;
 std::map<Key,int> delayed_keys;
 const int ARROW_SIMUL_FRAMES = 3;
 
+// Terrain map coordinates to simulate a click for 8-directional movement/waiting
+const int SOUTHWEST = 1;
```

Comment:
Pretty sure we already have these constants in an enum somewhere? Probably `eDirection`?

Reply:
oh, eDirection has

```
	DIR_N = 0,
	DIR_NE = 1,
	DIR_E = 2,
	DIR_SE = 3,
	DIR_S = 4,
	DIR_SW = 5,
	DIR_W = 6,
	DIR_NW = 7,
	DIR_HERE = 8,
```

But these don't correspond to indices of the terrain_click array.

Reply:
Is there a reason the terrain_click array has to be ordered that way? (I'm genuinely not sure, for the record.)

Reply:
Good question. I believe it's because of this:

https://github.com/calref/cboe/blob/336d0e626b23b50f6193a178ae3593b475cbf952/src/game/boe.actions.cpp#L1639-L1647

The numpad keys are in an array in ascending order,  and terrain_click is ordered to correspond with them.

Since I've copy/pasted the terrain_click array into `fire_delayed_key()` that instance of it could be reordered to correspond with `eDirection` values and remove the constants I made.

Reply:
I see, seems the intent was likely for `keypad[chr - '0']` to produce the keypad key for a given digit, and `terrain_click` is required to be sorted in the same order.

It's a bit ugly to have two copies of `terrain_click` with two different orders, but I guess that would work. I do suggest at least adding a TODO comment though.